setup.py

```diff
@@ -403,7 +403,7 @@ def run(self):
 
 def check_setuptools_for_dist():
     if "setuptools" not in sys.modules:
-        raise Exception("setuptools not available")
+        raise ModuleNotFoundError("setuptools not available")
```

Comment:
ModuleNotFoundError is raised if a module can't be loaded, which we are not checking here... but ymmv

Reply:
I was trying to fix that raising a generic Exception is not a good practice, because to catch it we have to catch all exceptions.

Maybe a ImportError would be more appropriate? I think we are checking that `import setuptools` will work, and that is the exception you would get if setuptools wasn't available.

Reply:
It was checking if setuptools was loaded without loading it (since that would monkey patch things). This was from when we had optional setuptools support, so I think the best is to just remove it that function.

Reply:
Thanks for helping to provide the context for why that function was there. I went ahead and removed the function.